extensions/arc/runtime/src/main/java/io/quarkus/arc/config/ConfigProperties.java

```diff
@@ -19,4 +21,47 @@
      * If the default is used, the class name will be used to determine the proper prefix
      */
     String prefix() default UNSET_PREFIX;
+
+    /**
+     * The naming strategy to use for the corresponding property. This only matters for fields or method names that contain
+     * both lower case and upper case characters.
+     *
+     * {@code NamingStrategy.DEFAULT} means that whatever the name of the field / method is, that will be the name of the
+     * property.
+     * {@code NamingStrategy.KEBAB_CASE} means that the name of property is derived by replacing case changes with a dash.
+     * For a example:
+     *
+     * /**
+     * 
+     * <pre>
+     * &#64;ConfigProperties(prefix="whatever")
+     * public class SomeConfig {
+     *   public fooBar;
+     * }
+     * </pre>
+     *
+     * Then to set the {@code fooBar} field, the corresponding property would be {@code whatever.fooBar}.
+     * If {@code namingStrategy=NamingStrategy.KEBAB_CASE} were being used, then the corresponding property would be
+     * {@code whatever.foo-bar}
+     */
+    NamingStrategy namingStrategy() default NamingStrategy.DEFAULT;
+
+    enum NamingStrategy {
+        DEFAULT {
+            @Override
+            public String getName(String name) {
+                return name;
+            }
+        },
+        KEBAB_CASE {
+            @Override
+            public String getName(String name) {
+                return StringUtil.hyphenate(name);
+            }
+        };
+
+        public String getName(String name) {
```

Comment:
This method can be declared as abstract

Reply:
Damn why didn't I think of that...

Reply:
I'll push a fix in a few hours

Reply:
Thanks for catching this, it should be fixed now